crates/interfaces/src/provider.rs

```diff
@@ -5,12 +5,27 @@ use reth_primitives::{
 use std::path::PathBuf;
 use thiserror::Error;
 
+/// Provider result type.
+pub type ProviderResult<Ok> = Result<Ok, ProviderError>;
+
+impl From<reth_nippy_jar::NippyJarError> for ProviderError {
+    fn from(err: reth_nippy_jar::NippyJarError) -> Self {
+        ProviderError::NippyJar(err.to_string())
+    }
+}
+
 /// Bundled errors variants thrown by various providers.
 #[derive(Clone, Debug, Error, PartialEq, Eq)]
 pub enum ProviderError {
```

Comment:
since this is the error type for all the provider traits, we should add an `Other(box dyn error)` variant so additional errors can be used by implementers

Reply:
shall we do the `String` instead. box dyn will require removing `PartialEq` & `Eq` which are useful in testing

Reply:
I see, let's skip for now, can always change later